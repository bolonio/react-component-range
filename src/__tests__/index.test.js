import React from "react";
import ReactRange from "../";
import renderer from "react-test-renderer";

test("ReactRange renders correctly with no props", () => {
  const component = renderer.create(<ReactRange />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("ReactRange renders disabled component", () => {
  const component = renderer.create(<ReactRange disabled={true} />);

  let tree = component.toJSON();
  expect(tree.props.disabled).toEqual(true);
});

test("ReactRange renders component with min and max", () => {
  const component = renderer.create(<ReactRange min={100} max={1000} />);

  let tree = component.toJSON();
  expect(tree.props.min).toEqual(100);
  expect(tree.props.max).toEqual(1000);
});
