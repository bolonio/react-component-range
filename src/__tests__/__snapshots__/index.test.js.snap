// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactRange renders correctly with no props 1`] = `
<input
  aria-valuemax={100}
  aria-valuemin={0}
  className="input-range"
  defaultValue={0}
  disabled={false}
  max={100}
  min={0}
  onChange={[Function]}
  required={false}
  step={1}
  type="range"
/>
`;
